File: src/main/java/io/teamif/rpgstats/enums/RoleType.java
package io.teamif.rpgstats.enums;

import lombok.AccessLevel;
import lombok.Getter;

public enum RoleType {

    Warrior("전사"),
    Rogue("도적"),
    Wizard("마법사");

    @Getter(AccessLevel.PUBLIC)
    private String roleName;

    RoleType(String roleName) {
        this.roleName = roleName;
    }
}

File: src/main/kotlin/io/teamif/rpgstats/role/RoleType.kt
package io.teamif.rpgstats.role

enum class RoleType(val roleName: String, val roleClass: Class<out AbstractRole>) {
    Warrior("전사", WarriorRole::class.java),
    Rogue("도적", RogueRole::class.java),
    Wizard("마법사", WizardRole::class.java);
}